File: apps/api/utils/parse_yaml.py
import yaml

def parse_yaml_to_dict(content: str) -> dict | None:
  content = content.strip()

  if content.startswith("```yaml"):
      content = content[7:].strip()
  elif content.startswith("```yml"):
      content = content[6:].strip()
  
  if content.endswith("```"):
      content = content[:-3].strip()

  yaml_content = content.replace("\\n", "\n")

  try:
    return yaml.safe_load(yaml_content)
  except yaml.YAMLError:
    return None

File: apps/web/app/utils/parseYaml.ts
import yaml from 'js-yaml';

export function parseYamlToDict(content: string): Record<string, any> | null {
  let str = content.trim();

  if (str.startsWith('```yaml')) {
    str = str.slice(7).trim();
  } else if (str.startsWith('```yml')) {
    str = str.slice(6).trim();
  }

  if (str.endsWith('```')) {
    str = str.slice(0, -3).trim();
  }

  str = str.replace(/\\n/g, '\n');

  try {
    return yaml.load(str) as Record<string, any>;
  } catch (err) {
    return null;
  }
}
